### DISCUSSION TITLE
Bun.input()

### DISCUSSION THREAD

**Discussion by bdotsamir** [oven-sh/bun#6191] [Q&A] [answered]
The built-in node.js readline module is clunky and weird to use. It's not impossible to understand, simply attach the module to process.stdin and wait for the `line` event to be emitted.

But that's weird, and frankly, other languages make it way easier. [Looking at you, Python.](https://docs.python.org/3/library/functions.html#input)

It would be great for Bun to expose an async `input()` method on the Bun namespace, maybe optionally providing a prompt. Once the user hits enter, the string the user entered is returned. 

I'm teaching a group of students on how to code using TypeScript, and in lieu of setting up tsc, I'm using Bun (which has been awesome so far!). I want them to be able to input things from the terminal, but it's weird as hell to have them set up the readline module with the event listener.. that's just one more thing I need to go over with them in the short time we have together in class.

**Comment by amytimed** [comment]
you could try `prompt()`, its not async though

**Comment by danielkaczmarczyk** [comment]
@bdotsamir what have you ended up using?

**Reply by bdotsamir** [reply, marked as answer]
I ended up using `prompt()`, since it seems to work just as fine in the console. Warning, it does block the thread. For me, this isn't an issue, but it may be for you.

**Reply by duanckham** [reply]
How do I hide secret input when using the `prompt()`?
